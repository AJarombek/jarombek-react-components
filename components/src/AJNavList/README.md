### Overview

Navigation list used in mobile drop-downs.

**Use cases:**

```javascript
import { AJNavList } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJNavList
    items={[
      {text: 'Home', link: '/'},
      {text: 'Profile', link: '/profile'}
    ]} />
```

### Files

| Filename                   | Description                                                               |
|----------------------------|---------------------------------------------------------------------------|
| `AJNavList.js`             | The reusable `AJNavList` component.                                       |
| `index.js`                 | Index file which exports the `AJNavList` component.                       |
| `styles.js`                | JSS styles for the `AJNavList` component.                                 |
