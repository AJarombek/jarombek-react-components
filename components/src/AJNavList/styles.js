/**
 * Stylesheet for the AJNavList component using JSS.
 * Author: Andrew Jarombek
 * Date: 2/18/2020
 */

export default {
  ajNavList: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center'
  },
  ajNavListItem: {
    width: '100%',

    '& > div': {
      display: 'flex',
      width: '90%',
      height: '40px',
      margin: '0 5%',
      borderBottom: '1px solid #ccc',

      textDecoration: 'none',
      color: '#777',
      transition: 'color 0.5s ease',
      cursor: 'pointer',

      '&:hover': {
        textDecoration: 'underline'
      },

      '& > div': {
        margin: '0 auto',
        alignSelf: 'center'
      }
    }
  },
  ajNavListLastItem: {
    '& > div': {
      borderBottom: 'none'
    }
  }
};
