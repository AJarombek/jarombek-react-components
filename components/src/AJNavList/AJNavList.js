/**
 * Navigation list used in mobile drop-downs.
 * @author Andrew Jarombek
 * @since 2/18/2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';
import classnames from 'classnames';
import {v4 as uuid} from 'uuid';

import styles from './styles';

const useStyles = createUseStyles(styles);

/**
 * Component representing a list commonly used by mobile dropdowns for the
 * website navigation header.
 * @param items Details about each item in the list.
 * @return {*} React elements that make up a navigation list.
 */
const AJNavList = ({items}) => {
  const classes = useStyles();

  if (!Array.isArray(items)) {
    items = [];
  }

  return (
    <div className={classnames(classes.ajNavList, 'aj-nav-list')}>
      { items.map((item, index) => {

        let listItemClasses;
        if (index === items.length - 1) {
          listItemClasses = classnames(classes.ajNavListItem, 'aj-nav-list-item',
            classes.ajNavListLastItem, 'aj-nav-list-last-item')
        } else {
          listItemClasses = classnames(classes.ajNavListItem, 'aj-nav-list-item')
        }

        return (
          <div key={uuid()} className={listItemClasses}>
            <div onClick={() => item.onClick()}>
              <div>{item.content}</div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

AJNavList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    content: PropTypes.node.isRequired,
    onClick: PropTypes.func.isRequired
  })).isRequired
};

export default AJNavList;
