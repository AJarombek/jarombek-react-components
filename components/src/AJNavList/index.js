import AJNavList from './AJNavList';

export default AJNavList;
