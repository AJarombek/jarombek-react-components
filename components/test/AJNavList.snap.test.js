import React from 'react';
import AJNavList from '../src/AJNavList/AJNavList';
import renderer from 'react-test-renderer';

/**
 * Snapshot test for the {@link ../../components/AJNavList} component.
 * @author Andrew Jarombek
 * @since 2/18/2020
 */

it('renders correctly', () => {
  const tree = renderer.create(
    <AJNavList
      items={[
        {content: 'Home', onClick: () => {}},
        {content: 'Profile', onClick: () => {}}
      ]}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
