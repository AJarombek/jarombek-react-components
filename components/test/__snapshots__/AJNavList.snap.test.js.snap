// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ajNavList-0-2-1 aj-nav-list"
>
  <div
    className="ajNavListItem-0-2-2 aj-nav-list-item"
  >
    <div
      onClick={[Function]}
    >
      <div>
        Home
      </div>
    </div>
  </div>
  <div
    className="ajNavListItem-0-2-2 aj-nav-list-item ajNavListLastItem-0-2-3 aj-nav-list-last-item"
  >
    <div
      onClick={[Function]}
    >
      <div>
        Profile
      </div>
    </div>
  </div>
</div>
`;
